sktime/forecasting/base/_fh.py

```diff
@@ -504,7 +514,11 @@ def to_indexer(self, cutoff=None, from_cutoff=True):
             Indexer.
         """
         if from_cutoff:
-            return self.to_relative(cutoff).to_pandas() - 1
+            try:
```

Comment:
should be timedelta based index, no?

Reply:
The more I think about it, the less sense this method (`to_indexer`) makes to me. The whole reason I want to extend to timdelta/dateoffset is to break the link between time series index and forecasting horizon. Let's say we have a time series with the following date index: `[Jan 1, Jan 2, Jan 2, Jan 5]`. The integer index here is `[0, 1, 2, 3]`. But if I start thinking in relative time-based terms, say `fh = 1 day`, and a `cutoff = Jan 2`, then indexing into the this time series is nothing but defining a time threshold and simply filtering out either values before or after this threshold, say `Jan 3`, or `y[y.index < cutoff + fh]`. In such arithmetic integer index is just not very helpful.

Reply:
I don't think filtering should play a role here, since in the deployment setting, the time stamps for which you forecast have not been yet observed, hence do not exist in the form of a `pd.DataFrame` with data in it, a-priori.

Reply:
exactly! I have the same argument in my head, just in different words, that blocks me from writing anything in this method. do you have a clear picture how this method should be used in case of timedelta horizons?

Reply:
hmm, what needs to be done/decided to resolve this? I'm slightly lost where this was going.

Reply:
I have already asked @aiwalter for his opinion on this matter. To me it looks like a completely new issue, implement `to_indexer` for timedelta horizons, which still does not make any sense to me. How about `NotImplementedError` for the time being and allowing me to move forward with my main idea?

Reply:
hm, yes.

Then, let's do as follows:
* kindly open an issue on this identified item
* here, kindly summarize your argument why we can skip this, and how we ensure downwards compatibility in the matter

Reply:
- https://github.com/alan-turing-institute/sktime/issues/2376
- In this PR the core change is the comparison with `pd.Timedelta(0)` in `sktime.forecasting.base._fh.ForecastingHorizon._is_in_sample`. This comparison allows me to move forward with generalizing splitters in `sktime.forecasting.model_selection._split` to work with non-integer windows, cutoffs, and forecasting horizons. `to_indexer` has nothing do with these changes. Hence, I would prefer to stay focused on my original plan here https://github.com/alan-turing-institute/sktime/issues/1737.

Reply:
this is the one, #2377, right?

Reply:
Correct, the new issue is #2376